### PR TITLE
Remix button progress spinner & banner message showing that user has remixed

### PR DESCRIPTION
**Pull request by benjiwheeler**
### Resolves:

With https://github.com/LLK/scratch-gui/pull/3965 , resolves https://github.com/LLK/scratch-www/issues/2328

### Changes:

Adds new banner to project page that congratulates user on remixing project

show a green banner right after they've remixed:

image

...and then on subsequent visits show the regular unshared banner:

image

Also, show remixing progress state in the Remix button:

dec-05-2018 17-41-10

And show a tooltip:

image

### CODE REVIEW

**Comment by benjiwheeler** [comment]
Made follow-on issue for counterclockwise spin: https://github.com/LLK/scratch-www/issues/2411

File: src/views/preview/presentation.jsx [hunk 1]
```diff
@@ -136,6 +139,18 @@ const PreviewPresentation = ({
                 message={embedCensorMessage(visibilityInfo.message)}
             />);
         }
+    } else if (justRemixed && isNewScratcher) {
+        banner = (
```
Comment (chrisgarrity) [inline src/views/preview/presentation.jsx]: Should the just remixed banner show for all users, or just new Scratchers? it looks like it'll only be newScratchers

File: src/views/preview/project-view.jsx [hunk 2]
```diff
@@ -103,11 +106,20 @@ class Preview extends React.Component {
...
+            if (this.state.justShared) {
+                this.setState({ // eslint-disable-line react/no-did-update-set-state
+                    justShared: false
+                });
+            }
         }
         if (this.state.projectId === '0' && this.state.projectId !== prevState.projectId) {
             this.props.resetProject();
-            this.setState({justShared: false}); // eslint-disable-line react/no-did-update-set-state
+            if (this.state.justRemixed || this.state.justShared) {
```
Comment (chrisgarrity) [inline src/views/preview/project-view.jsx]: is this `if` needed, or should it always reset these states (like the project)

Comment (benjiwheeler) [reply src/views/preview/project-view.jsx]: Unfortunately, `setState` will cause a re-render even if the values you set it to are identical to the current values. So this just avoids an unnecessary re-render, but will always re-render if `justRemixed` or `justShared` were true.

see https://itnext.io/react-setstate-usage-and-gotchas-ac10b4e03d60
image

File: src/views/preview/presentation.jsx (hunk 1 above)
Comment (benjiwheeler) [reply src/views/preview/presentation.jsx]: You're right, checked with designers, fixed

**Pull request merged**